zigpy/ota/provider.py

```diff
@@ -324,7 +324,13 @@ async def fetch_image(self) -> Optional[OTAImage]:
             with tarfile.open(fileobj=img_tgz) as tar:  # Unpack tar
                 for item in tar:
                     if item.name.endswith(".ota"):
-                        file_bytes = tar.extractfile(item).read()
```

Comment:
I think this would work better (if you don't just want to ignore this error):

```Python
tar_f = tar.extractfile(item)
assert tar_f is not None
file_bytes = tar_f.read()
```

The conditions for `None` being returned are pretty specific and probably will never be encountered:

```python
# If there's no data associated with the member (directory, chrdev,
# blkdev, etc.), return None instead of a file object.
```

Reply:
I've seen a static code checker complain about assertions - they're not compiled in optimized code.
I agree it's likely never encountered - a side effect of static code checks, but I prefer fixing it in the code rather than adding an exception for the tool.

Reply:
Sure. In that case, `raise ValueError(f"Issue ...` should be enough. It will automatically be logged as an exception here: https://github.com/zigpy/zigpy/blob/e91ac25d968403fd8e26c828e2177cfa2797973a/zigpy/util.py#L68

Reply:
Ok, changed.